src/main/java/request/HttpRequestBuilder.java

```diff
@@ -1,41 +1,27 @@
 package com.td.HttpServer;
 
-import java.util.*;
+import java.util.HashMap;
 import java.io.IOException;
 
 public class HttpRequestBuilder implements IRequestBuilder {
+  IRequestReader reader;
+  IRequestParser parser;
 
-  private IValidator requestVerifier;
-  public HttpRequestBuilder(IValidator requestVerifier) {
-    this.requestVerifier = requestVerifier;
+  public HttpRequestBuilder(IRequestReader reader, IRequestParser parser ) {
+    this.reader = reader;
+    this.parser = parser;
   }
 
-  public HttpRequest createRequest(String rawRequest) throws InvalidHttpRequestException {
+  public HttpRequest createRequest(IMessageIO client) throws InvalidHttpRequestException, IOException {
     HttpRequest request;
-    String requestLine = parseRequestLine(rawRequest);
-    HashMap<String, String> headers = parseHeaders(rawRequest);
+    String rawRequest = reader.getMessage(client);
+    String requestLine = parser.parseRequestLine(rawRequest);
+    HashMap<String, String> headers = parser.parseHeaders(rawRequest);
```

Comment:
Would it be okay for the `HttpRequestBuilder` to Contain the parsing logic and also be injected with a `SocketIO` that is asks to get a String and Bytes(if needed, because there is a body)??? 
Not sure if those are two different responsibilities. I don't think they are..